sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +633,24 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(javaOptions: Seq[String]): Option[VirtualJSFile] = {
+    val javaSysPropsPattern = "-D([^=]*)=(.*)".r
+    val formattedProps = javaOptions.collect {
+      case javaSysPropsPattern(propName, propValue) =>
+        "\"" + escapeJS(propName) + "\": \"" + escapeJS(propValue) + "\""
+    }
```

Comment:
I am actually not that sure. This method fills in the `System.getProperties()` map. It takes the `javaOptions` that are of interest to it. There *could* be another task/setting somewhere, that would take other parts of `javaOptions` and do other things with it. For example, one could imagine capturing `-Xmx` options and reroute them as `args` to `NodeJSEnv`.

Various `javaOptions` could have to be handled by various parts of our setting pipeline, and/or users' builds. The extraction of system properties should not break everything else.